src/classes/RD2_ScheduleService.cls

```diff
@@ -739,19 +791,19 @@ public without sharing class RD2_ScheduleService {
     }
 
     /***
-    * @description Adds recurring donation schedules to map by recurring donation Id
-    * @param rd SObject
-    * @param newSchedulesByRdId Map<Id, List<RecurringDonationSchedule__c>>
+    * @description Adds RD Schedule to the RD schedules list if the schedule is active
+    * @param schedule Recurring Donation Schedule
+    * @param newSchedulesByRdId Schedules mapped by the RD Id
     * @return void
     */
-    private void addUpdatedSchedules(RecurringDonationSchedule__c rd, Map<Id, List<RecurringDonationSchedule__c>> newSchedulesByRdId) {
+    private void addScheduleIfActive(RecurringDonationSchedule__c schedule, Map<Id, List<RecurringDonationSchedule__c>> newSchedulesByRdId) {
 
-        if (rd.EndDate__c == null || (rd.EndDate__c >= currentDate && rd.StartDate__c < rd.EndDate__c)) {
+        if (isWithinDateRange(schedule, currentDate)) {
```

Comment:
@npsp-reedestockton , @force2b 
The previous verification was for `rd.StartDate__c < rd.EndDate__c` while the `isWithinDateRange()` checks for `rd.StartDate__c <= rd.EndDate__c` that is required to return an active pause that skips one installment. Is there an issue to using `isWithinDateRange()` ?

Reply:
Good question. The `Active__c` formula defines active as `StartDate__c <= EndDate__c`, so I think the change the logic here is line with that.

Reply:
Yes.  I don't understand why it was ever < EndDate__c.  My guess is that this was an error which is now be fixed.